diff --git a/net/url_request/url_request.h b/net/url_request/url_request.h
index 90e90f42a18d527afc44f553711f9eed773518b8..1fe6b0ccd73d87ea771129ac49f635dc3bf01ada 100644
--- a/net/url_request/url_request.h
+++ b/net/url_request/url_request.h
@@ -335,6 +335,14 @@ class NET_EXPORT URLRequest : public base::SupportsUserData {
     force_ignore_site_for_cookies_ = attach;
   }
 
+
+  bool is_ad_tracker_tagged() const {
+    return is_ad_tracker_tagged_;
+  }
+  void set_is_ad_tracker_tagged(bool tagged) {
+    is_ad_tracker_tagged_ = tagged;
+  }
+
   // Indicates if the request should be treated as a main frame navigation for
   // SameSite cookie computations.  This flag overrides the IsolationInfo
   // request type associated with fetches from a service worker context.
@@ -1084,6 +1092,8 @@ class NET_EXPORT URLRequest : public base::SupportsUserData {
   // Unpartitioned cookies are unaffected by this field.
   std::optional<CookiePartitionKey> cookie_partition_key_ = std::nullopt;
 
+  bool is_ad_tracker_tagged_ = false;
+
   bool force_ignore_site_for_cookies_ = false;
   bool force_main_frame_for_same_site_cookies_ = false;
   bool is_shared_resource_ = false;
