diff --git a/services/network/url_loader.cc b/services/network/url_loader.cc
index b4e32dd34c6d5f826da777dce7502da7ff75e76a..a2525d09d07735666ed33c71a278203caf7424e5 100644
--- a/services/network/url_loader.cc
+++ b/services/network/url_loader.cc
@@ -446,6 +446,11 @@ URLLoader::URLLoader(
       permissions_policy_(request.permissions_policy) {
   DCHECK(delete_callback_);
 
+  bool ad_tracker_tagged = false;
+  if (request.headers.HasHeader("sec-X-AdTracker")) {
+    ad_tracker_tagged = true;
+  }
+
   if (options_ & mojom::kURLLoadOptionReadAndDiscardBody) {
     if (!factory_params_->is_orb_enabled) {
       discard_buffer_ =
@@ -480,6 +485,9 @@ URLLoader::URLLoader(
   // Set up UserData (pointing to `this`) first.
   url_request_->SetUserData(kUserDataKey,
                             std::make_unique<UnownedPointer>(this));
+
+  url_request_->set_is_ad_tracker_tagged(ad_tracker_tagged);
+
   // Configure the main request parameters. This must happen after setting
   // UserData, as `ConfigureUrlRequest` might internally retrieve data (e.g.,
   // PermissionsPolicy) via the `url_request_`'s UserData pointer.
@@ -2052,6 +2060,11 @@ void URLLoader::SetRawRequestHeadersAndNotify(
           network::mojom::HttpRawHeaderPair::New();
       pair->key = header.first;
       pair->value = header.second;
+      if (header.first == "sec-X-AdTracker") {
+          LOG(ERROR) << "sec-X-AdTracker header found in raw request headers: "
+                     << header.second;
+          continue; 
+      }
       header_array.push_back(std::move(pair));
     }
     DispatchOnRawRequest(std::move(header_array));
